pkgs/development/node-packages/default.nix

```
@@ -181,6 +181,21 @@ let
       ];
     };
 
+    vega-lite = super.vega-lite.override {
+        # npx tries to install vega from scratch at vegalite runtime if it
+        # can't find it. We thus replace it with a direct call to the nix
+        # derivation. This might not be necessary anymore in future vl
+        # versions: https://github.com/vega/vega-lite/issues/6863.
+        postInstall = ''
+        substituteInPlace $out/lib/node_modules/vega-lite/bin/vl2pdf \
+          --replace "npx -p vega vg2pdf"  "${self.vega-cli}/bin/vg2pdf"
+        substituteInPlace $out/lib/node_modules/vega-lite/bin/vl2svg \
+          --replace "npx -p vega vg2svg"  "${self.vega-cli}/bin/vg2svg"
+        substituteInPlace $out/lib/node_modules/vega-lite/bin/vl2png \
+          --replace "npx -p vega vg2png"  "${self.vega-cli}/bin/vg2png"
+    '';
```

```suggestion
          substituteInPlace $out/lib/node_modules/vega-lite/bin/vl2pdf \
            --replace "npx -p vega vg2pdf"  "${self.vega-cli}/bin/vg2pdf"
          substituteInPlace $out/lib/node_modules/vega-lite/bin/vl2svg \
            --replace "npx -p vega vg2svg"  "${self.vega-cli}/bin/vg2svg"
          substituteInPlace $out/lib/node_modules/vega-lite/bin/vl2png \
            --replace "npx -p vega vg2png"  "${self.vega-cli}/bin/vg2png"
      '';
```

Sorry for the nitpick, but it should be perfect now!